app/controllers/IncomeController.scala

```diff
@@ -264,20 +261,15 @@ class IncomeController @Inject()(
           case Right(cache) =>
             val incomeName :: newAmount :: incomeId :: incomeType :: Nil = cache.toList
 
-            journeyCacheService.flush().flatMap { _ =>
-              taxAccountService.updateEstimatedIncome(
-                user.nino,
-                FormHelper.stripNumber(newAmount).toInt,
-                TaxYear(),
-                incomeId.toInt) flatMap {
-                case TaiSuccessResponse =>
-                  estimatedPayJourneyCompletionService.journeyCompleted(incomeId).map { _ =>
-                    respondWithSuccess(incomeName, incomeId.toInt, incomeType, newAmount)
-                  }
-                case failure: TaiFailureResponse =>
-                  throw new RuntimeException(s"Failed to update estimated income with exception: ${failure.message}")
-              }
-            }
+            for {
```

Comment:
How come you even need to do this for-yield when all the values that are 'yielded' already exist in `val incomeName :: newAmount :: incomeId :: incomeType :: Nil = cache.toList`